libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -14,25 +14,189 @@
 
 #include "spvcir_pass.h"
 
-#include <fstream>
+#include "source/opt/ir_context.h"
 
 namespace spvtools {
 namespace opt {
 
+SpvcIrPass::SpvcIrPass(spirv_cross::ParsedIR &ir) {
+  ir_ = &ir;
+  current_function_ = nullptr;
+  current_block_ = nullptr;
+
+  auto s = ir_->spirv.data();
+  uint32_t bound = s[3];
+  ir_->set_id_bounds(bound);
+}
+
 Pass::Status SpvcIrPass::Process() {
-  // A temporary pass.
-  // TODO (sarahM0): Replace this by a switch-case that generates a spirv-cross
-  // IR
-  uint32_t options = 0;
-  std::ostringstream oss;
-  get_module()->ForEachInst([&oss, options](const Instruction* inst) {
-    oss << inst->PrettyPrint(options);
-    if (!IsTerminatorInst(inst->opcode())) {
-      oss << std::endl;
-    }
-  });
+  get_module()->ForEachInst(
+      [this](Instruction *inst) { genSpirvCrossIR(inst); });
+
+  assert(!current_function_ && "Function was not terminated.");
```

Comment:
It's ok to assert here because this class assumes the module passes validation. Suggest adding a comment to that effect.